### ISSUE TITLE
Calico-Node clashes with Aqua's Drift-Protection runtime control. Is this expected behaviour? 

### ISSUE THREAD

**Issue by MarkTopping** [projectcalico/calico#6624]
In an Azure AKS environment where Calico is enabled as the Network Policy provider and within a cluster where AquaSec is used to enhance cluster security, Aqua is repeatedly blocking Calico-Node from performing an action that it considers unsafe.

Within Aqua, we are using a Runtime control labelled as 'Drift Prevention'. It's purpose is to prevent a container from running code that was not present in the original image. 

A blog which speaks more to this feature can be found here: https://blog.aquasec.com/cloud-native-security-drift-prevention

One assumes (or hopes) that this is somehow inhibiting Calico from doing it's job. 

I'd like to ask if anybody understand:
- what Calico is trying to achieve when it runs this process? 
- is it expected and necessary / will Aqua blocking this process impede the function of the product?
- Is Aqua mis-firing / mis-reporting an issue? Or does Aqua have a good point - Is Calico doing something here that Aqua is rightly preventing because of the security risk it represents? 

I can raise an issue with AquaSec if calico is actually not performing actions which violate the control.
I can give Calico an exemption from the policy if it's behaviour is justified. 

EDIT
I've been running Calico and Aqua for several months harmoniously. This issue only started occurring on 24th August onwards. 

## Current Behavior
Aqua blocks calico-node pods from running "./run" approximately 450 times per hour per node. 

The process reported to be executing the command is /usr/local/bin/runsvc 

EDIT
Within the calico-node pods the following logs can be observed:
- runsv node-status-reporter: fatal: unable to start log/.run: access denied
- runsv monitor-addresses: fatal: unable to start ./run: access denied
- runsv felix: fatal unable to start ./run: access denied
- runsv felix: fatal unable to start log/./run: access denied

## Your Environment
* Calico version: 3.21.6
* Orchestrator version: AKS 1.23.8 - with AquaSec deployed and Calico Network Policy enabled
* Operating System and version: Ubuntu

**Comment by caseydavenport** [comment]
I'm not sure why aqua is flagging `run` as code outside of the calico/node filesystem, as those are all contained within the calico/node container image.

Calico obviously interacts with the host kernel out of necessity, but to my knowledge it shouldn't be calling userspace binaries that aren't packaged withing calico/node itself. 

[calico/node does copy the `run` files within the container filesystem](https://github.com/caseydavenport/calico/blob/62f43c08beced4f1d65c2c939ede14bcf1208dda/node/filesystem/etc/rc.local#L59-L62) as part of its startup - perhaps Aqua is confused by that?

**Comment by lwr20** [comment]
> This issue only started occurring on 24th August onwards.
Did either Calico or Aqua get updated in your cluster on that date?

**Comment by MarkTopping** [comment]
@lwr20 I only know part of the story here. Calico is deployed/updated within the cluster by Microsoft so I don't have visibility on whether or not they pushed out a new release during this time - it wasn't something I was tracking. I did assume an update occurred though given the sudden clash with Aqua

Aqua has a client and a server side element. I know that the client side (which runs adjacent to the Calico components) wasn't updated since I control the release cadence there; and I know that the service side wasn't due an update but then I'm not in control of that either so it's possible that it was. The detection happens client side though which is why I got the impression that something had changed within Calico rather than vice-versa.

@caseydavenport OK - thanks for that information.  If I end up raising a ticket with Aqua I'll cite this.
